app/uk/gov/hmrc/securemessage/services/SecureMessageService.scala

```diff
@@ -19,51 +19,83 @@ package uk.gov.hmrc.securemessage.services
+      res <- emailParticipants(cnv)
+    } yield res).value
 
-  def createConversation(conversationRequest: ConversationRequest, client: String, conversationId: String)(
+  @SuppressWarnings(Array("org.wartremover.warts.Equals"))
+  val hasEmail: Participant => Boolean = p =>
+    p.participantType == PSystem || (p.participantType == PCustomer && p.email.isDefined)
+
+  @SuppressWarnings(
+    Array(
+      "org.wartremover.warts.Nothing",
+      "org.wartremover.warts.Any",
+      "org.wartremover.warts.Equals",
+      "org.wartremover.warts.ImplicitParameter"))
+  private def emailParticipants(cnv: Conversation)(
     implicit hc: HeaderCarrier,
-    ec: ExecutionContext): Future[Result] = {
-    val conversation = conversationRequest.asConversation(client, conversationId)
-    repo.insertIfUnique(conversation).map { isUnique =>
-      if (isUnique) {
-        getEmailRequest(conversationRequest) match {
-          case Left(err) => BadRequest(err)
-          case Right(req) =>
-            val _ = emailConnector.send(req)
-            Created
+    ec: ExecutionContext): EitherT[Future, SecureMessageException, Int] = {
+    val fEmails: EitherT[Future, SecureMessageException, List[EmailAddress]] = {
+      val filteredEmails: List[List[EmailAddress]] =
+        for (p <- cnv.participants if p.participantType == PCustomer && p.email.isDefined) yield p.email.toList
+      val emails =
+        filteredEmails match {
+          case List() =>
+            Left(new SecureMessageException { val message = "Verified email address could not be found" })
+          case emails => Right(emails.flatten)
         }
-      } else {
-        Conflict("Duplicate of existing conversation")
-      }
+      EitherT(Future.successful(emails))
     }
+    for {
+      emails <- fEmails
+      res    <- EitherT(emailConnector.send(EmailRequest(emails, cnv.alert.templateId))).leftWiden[SecureMessageException]
+    } yield res
   }
 
-  private def getEmailRequest(request: ConversationRequest): Either[String, EmailRequest] = {
-    val emailAddresses = request.recipients.flatMap(r => r.customer.email.toList)
-    if (emailAddresses.nonEmpty) {
-      Right[String, EmailRequest](EmailRequest(emailAddresses, request.alert.templateId))
-    } else {
-      Left[String, EmailRequest]("No recipient email addresses provided")
-    }
-  }
+  @SuppressWarnings(Array("org.wartremover.warts.Equals", "org.wartremover.warts.ImplicitParameter"))
+  private def getConversationWithEmailFromCds(
+    cnv: Conversation)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Conversation] =
+    Future
+      .sequence(cnv.participants.map(p => {
+        if (p.participantType == PSystem || (p.participantType == PCustomer && p.email.isDefined)) {
+          Future.successful(p)
+        } else {
+          channelPrefConnector
+            .getEmailForEnrolment(p.identifier)
+            .map(
+              _.fold(_ => p, e => p.copy(email = Some(e)))
```

Comment:
so here you will lose the unsuccessful lockups ?

Reply:
Here I return the original participant with no email